ddtrace/propagation/http.py

```diff
@@ -60,6 +70,30 @@ def parent_call():
         if span_context.dd_origin is not None:
             headers[HTTP_HEADER_ORIGIN] = str(span_context.dd_origin)
 
+        # Only propagate tags that start with `_dd.p.`
+        tags_to_encode = {}  # type: Dict[str, str]
+        for key, value in span_context._meta.items():
+            # DEV: encoding will fail if the key or value are not `str`
+            key = ensure_str(key)
+            if key.startswith("_dd.p."):
+                tags_to_encode[key] = ensure_str(value)
+
+        if tags_to_encode:
+            encoded_tags = None
+
+            try:
+                encoded_tags = encode_tagset_values(tags_to_encode)
+            except TagsetMaxSizeError:
+                # We hit the max size allowed, add a tag to the context to indicate this happened
+                span_context._meta["_dd.propagation_error"] = "max_size"
```

Comment:
Two questions: 

1. Can we maintain the namespace of `_dd.p.` for errors?
2. Could we follow the pattern of errors for span data and record `exc_info` in its entirety? That could be helpful to have when such errors occur rather than expecting them in logs.

Reply:
1. From the spec we are being asked to keep as `_dd.propagation_error`, we also do not want to propagate to other services, which is why we want to avoid `_dd.p.` (prefix used to indicate that a tag should be propagated).
2. That is an interesting idea. I wonder if my idea of raising an exception for these issues is going to be common implementation pattern or not? The original idea was to have a specific/static sentinel value that is shared by all implementations.

Reply:
2. Also, just because we cannot propagate them to other services, we'll still be adding them to the chunk root span for the trace. So we'd end up with `_dd.propagation_error: max_size`, and then we'll be able to look at the `_dd.p.*` tags on that span.

It isn't the fully traceback/error info, but we _should_ have enough data there to know an error occurred, what type of error, and the data that produced the error.

Reply:
1. Makes sense to avoid the `_dd.p` namespace if it's intended to always be propagated.
2. Right, the original intent should be met, the additional exception information would be useful if issues.

In thinking through the implications of this implementation, multiple http within the same tracing context would attempt to encode the same tag set and fail again, is that correct?

Reply:
> In thinking through the implications of this implementation, multiple http within the same tracing context would attempt to encode the same tag set and fail again, is that correct?

this is correct, we could maybe save some headache if we check the context for `_dd.propagation_error` first and then try to propagate?